### ISSUE TITLE
Resolve USB debugging auth prompt on first boot

### ISSUE THREAD

**Issue by patcon** [patcon/mission-impossible-android#19]
Even though USB debugging appears to be disabled when I go to check, if my device is still plugged into my computer after the whole `make build_deploy` process, a dialog pops up asking me to authorize the device, showing a fingerprint. I would never auth it, but it's odd, for sure

**Comment by patcon** [comment]
fwiw, clicking "Yes" still doesn't provide adb access, which would be worrying.

**Comment by SchnWalter** [comment]
It might have something to do with the CM build type. On my device I have `ro.build.type=userdebug` and according to https://github.com/CyanogenMod/android_build/blob/cm-12.0/core/build-system.html#L278-L291 `adb` is enabled by default

And I'm not sure, but we migth need to look over the `/init.usb.rc` file: https://gist.github.com/SchnWalter/f6a8ddc783d3938b8620

**Comment by SchnWalter** [comment]
While working on properly implementing logs in the init scripts, I have come across a couple of fixes for this issue.

The first solution uses `/system/bin/settings`, which actually runs [java code](https://github.com/CyanogenMod/android_frameworks_base/blob/cm-12.1/cmds/settings/settings).
```bash
# Disable android USB debugging using ADB.
# @see https://developer.android.com/reference/android/provider/Settings.Global.html#ADB_ENABLED
ADB_STATUS=`settings get global adb_enabled`
$LOG "ADB status is: ${ADB_STATUS}"
$LOG "Setting ADB status (adb_enabled) to 0."
settings put global adb_enabled 0
ADB_STATUS=`settings get global adb_enabled`
$LOG "ADB status is: ${ADB_STATUS}"
```
This is all good as long as you have the JRE up and running, we currently have it because the current init script is using some weird loop and is waiting for a process called `com.android.systemui` to show up, but I would like to avoid all those hacks that might cause all kind issue.

The seccond (proper) option is to set the `persist.sys.usb.config` property to `0`, and this works before JRE is available and it also changes `adb_enabled` to `0`.

```bash
# Disable android USB debugging using ADB.
USB_CONFIG=$(getprop sys.usb.config)
$LOG "The sys.usb.config is: ${USB_CONFIG}"

# The `sys.usb.config` property has the following format: foo,adb,bar.
# In order to disable ADB, it needs to be removed from the `sys.usb.config` property.
# You can run this command on the device to see possible values:
#   $ grep sys.usb.config /*.usb.rc
USB_CONFIG=$(echo ${USB_CONFIG} | sed -e 's/,\?adb,\?/,/g' -e 's/^,//' -e 's/,$//')

$LOG "Disabling android USB debugging using ADB."
setprop persist.sys.usb.config ${USB_CONFIG}
USB_CONFIG=`getprop sys.usb.config`
$LOG "The sys.usb.config is: ${USB_CONFIG}"
```

I'm probably going to create a PR with a lot of changes to the init scripts, but I'm going to try to document as much as possible.

**Comment by patcon** [comment]
Yes, definitely document. Ideally, PR's would be granular and not mega-PR's that change all the scripts. I know I've been MIA, but it's still very important to me that I can confidently vouch for this repo :)

I'm in SF, and have had some conversations that make me want to jump back into MIA and related projects, so hopefully I'll be more active within the month :)

Thanks @SchnWalter!

**Comment by SchnWalter** [comment]
I'm not sure exactly what has changed, but it looks like you are no longer prompted to authorize USB debugging on first boot. It's either related to the cleanup from #106 or to the fact that I have upgraded ADB from `v1.0.31` to `v1.0.32`...
